testomania/src/main/java/com/earth/testomania/home_screen/presentation/HomeScreen.kt

```diff
@@ -79,7 +91,42 @@ fun HomeScreen(
             )
         }
     }
+}
 
+@Composable
+fun SheetLayout(
+    modalBottomSheetState: ModalBottomSheetState,
+    scope: CoroutineScope,
+    navigator: DestinationsNavigator?
+) {
+    val viewModel: HomeScreenViewModel = hiltViewModel()
+    val pageType by viewModel.bottomSheetPageState.collectAsState(initial = BottomSheetScreen.Technical)
+    BottomContent(pageType, modalBottomSheetState, scope, navigator)
+}
+
+@Composable
+fun BottomContent(
+    pageType: BottomSheetScreen,
+    modalBottomSheetState: ModalBottomSheetState,
+    scope: CoroutineScope,
+    navigator: DestinationsNavigator?
+) {
+    when (pageType) {
+        is BottomSheetScreen.Technical -> CategorySelectorBottomSheet(
+            modalBottomSheetState,
+            scope
+        ) { quizCategory ->
+            scope.launch {
+                modalBottomSheetState.snapTo(ModalBottomSheetValue.Hidden)
+                //snapTo is quicker then hide
+            }.invokeOnCompletion {
```

Comment:
why not have `navigator?.navigate(TechnicalTestsScreenDestination(quizCategory))` inside launch {} ?

Reply:
it's a bug fix by @Nodrex as I guess, the bottomSheet stays on screen otherwise

Reply:
Yes, @kdiakonidze  is right, I fixed `bottomSheet stays on screen` Bug with this solution. 
If you @shalva97 have better solution please suggest

Reply:
as I check it works without "invokeOnCompletion" block. just calling navigation after bottomSheet hide method in launch block. Any objections on that @Nodrex ?

Reply:
It did not always worked for me

Reply:
ok, lets stay on solution by Nodar